### PR TITLE
Fix Uncaught Failures

### PR DESCRIPTION
**Pull request by brandonpage**
Also updated xcode & android api version + associated dependencies.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -21,12 +21,29 @@ aliases:
       sudo make && sudo make install
       git --version
 
+  - &downgrade-cocoapods
+    name: Downgrade to CocoaPods 1.4
```
Comment (wmathurin) [inline .circleci/config.yml]: 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -61,23 +78,43 @@ jobs:
...
+            ./test/test_force.js --os=android --apptype=hybrid_local
+            if [ ! -e tmp**/App_hybrid_local_android/platforms/android/app/build/outputs/apk/debug/app-debug.apk ]; then
+              exit 1
+            fi
           when: always
       - run:
           name: Build Hybrid Remote App
-          command: ./test/test_force.js --os=android --apptype=hybrid_remote
+          command:  |
+            ./test/test_force.js --os=android --apptype=hybrid_remote
```
Comment (wmathurin) [inline .circleci/config.yml]: So we can't work off the process exit code?

Comment (brandonpage) [reply .circleci/config.yml]: I tried that for a bit with mixed results before realizing we shouldn't really care about the process exist code for a test script.  The point of the automation is to test generating and building the app.  So why not base pass/fail on whether we successfully built the .app or apk?

Comment (wmathurin) [reply .circleci/config.yml]: That’s a good approach.

**Pull request merged**